lib/backbeat/Metrics.js

```diff
@@ -178,6 +178,8 @@ class Metrics {
      * @return {undefined}
      */
     getCompletions(details, cb, data) {
```

Comment:
I think ideally this can be broken into a scope per extension. That way we can avoid having these various conditions that may only grow in time—for example, we would would be following a similar pattern when introducing lifecycle metrics. I think, at least, the JSDoc could be updated to reflect that it is no longer serving just replicated stats.

Reply:
Sounds good, so create separate helper methods to handle each extension response case?

Reply:
I see, like separate entry methods for each extension, and then they can use the same helper methods for finding relevant data? I think this approach would be more straightforward for sure and we can add additional routes in a more clear and cleaner fashion.

I know my post review changes don't address this, but for now, it's more readable than current implementation.

Reply:
👍 Thanks for that!